code/controllers/subsystem/timer.dm

```diff
@@ -289,11 +289,20 @@ var/datum/subsystem/timer/SStimer
 	prev = null
 	return QDEL_HINT_QUEUE
 
-proc/addtimer(datum/callback/callback, wait, flags)
+var/list/addtimer_overhead_calls
+
+proc/addtimer(datum/callback/callback, wait, flags, suppress_zero_warning = FALSE)
 	if (!callback)
 		return
 
+
 	if (wait <= 0)
+		if(!suppress_zero_warning)
```

Comment:
If anything, what this should do is just add it to the queue to run next tick.

There is a technical difference to running something after whats currently running is finished, and running something now but don't let it's sleeps block me bro.

spawn(0) did the first, set waitfor only does the second one.

Somebody could want to do the first one with addtimers.